cargo-dist/src/tasks.rs

```diff
@@ -1524,6 +1537,69 @@ impl<'pkg_graph> DistGraphBuilder<'pkg_graph> {
         );
     }
 
+    fn add_omnibor_artifact_id_file(&mut self, to_release: ReleaseIdx) {
+        if !self.global_artifacts_enabled() || !self.inner.config.builds.omnibor {
+            return;
+        }
+
+        let release = self.release(to_release).clone();
+        let global_artifacts = release.global_artifacts;
+        let local_artifacts: Vec<_> = release
+            .variants
+            .into_iter()
+            .flat_map(|vidx| self.variant(vidx).local_artifacts.clone())
+            .collect();
+        let artifacts: Vec<_> = global_artifacts
+            .into_iter()
+            .chain(local_artifacts)
+            .map(|a| self.artifact(a).clone())
+            .collect();
+
+        for artifact in &artifacts {
+            let id = artifact.id.clone();
+            let src_path = artifact.file_path.clone();
+            match &artifact.kind {
```

Comment:
Nit: if this is written as an `if matches!(artifact.kind, ...)` condition, you could skip the unwanted `ArtifactKind` cases and only handle the desired ones.

Reply:
This was actually deliberate, because it means if another `ArtifactKind` is added the compiler will tell us we need to decide where it goes, instead of just assuming all new `ArtifactKind`s aren't relevant.

If you think I should change it anyway, lmk and I will. Otherwise I'll add a comment explaining why I did it.

Reply:
Ah, good call! That makes sense.